From 20a4a37269734b90aeea81ae6a24dddfda8ad5ab Mon Sep 17 00:00:00 2001
From: Jean Christophe Beyler <jean.christophe.beyler@intel.com>
Date: Mon, 25 Mar 2013 14:36:13 -0700
Subject: Dalvik: Fix Dalvik Klokwork Issue with libenc

BZ: 95586

Fixed an issue of an array overflow detected by Klokwork:

  - Libenc has an array of 44 entries
  - Back-end explicitly now does not go towards the libenc in cases where an overflow could happen

Category: device-enablement
Domain: AOSP-Dalvik-Compiler-CG
Origin: internal
Upstream-Candidate: no, needs rework

Change-Id: I0bc76e56388a3d9e45fca78a113cf5f5b856101b
Orig-MCG-Change-Id: I5c6220122995d20bf39db4c70904b5f27563b45d
Signed-off-by: Jean Christophe Beyler <jean.christophe.beyler@intel.com>
Signed-off-by: Qiming Shi <qiming.shi@intel.com>
Signed-off-by: Serguei Katkov <serguei.i.katkov@intel.com>
---
 vm/compiler/codegen/x86/RegisterizationBE.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/vm/compiler/codegen/x86/RegisterizationBE.cpp b/vm/compiler/codegen/x86/RegisterizationBE.cpp
index 2e0eb8e..0634892 100644
--- a/vm/compiler/codegen/x86/RegisterizationBE.cpp
+++ b/vm/compiler/codegen/x86/RegisterizationBE.cpp
@@ -771,7 +771,7 @@ static bool moveToChildPhysical (AssociationTable &associationsToUse,
         PhysicalReg targetReg = childVRToPhysicalReg[VR];
 
         //This should never happen but it will make buffer overflow checkers happy
-        if (targetReg >= PhysicalReg_SCRATCH_1) {
+        if (targetReg >= PhysicalReg_Null) {
             continue;
         }
 
-- 
1.7.4.1

